### ISSUE TITLE
Data over 30 days ago ignored

### ISSUE THREAD

**Issue by VelocityRa** [aardappel/procrastitracker#38]
Hello, thank you for this very useful piece of software.

I've been experiencing a bug:
**The database files are there, yet procrastitracker doesn't read anything before ~30 days.** (I did set it to "All Time").
I remember a while ago it wouldn't read anything before 2 to 3 months, I think. Now it's 30 days.

I've built it locally, if you want me to use the debugger for anything.

Thanks.

**Comment by aardappel** [comment]
That sounds weird. Not a bug I've heard of before, nor something that I have a hunch for what code could cause it.

Actually, I have one hunch: there's code that prunes part of the DB on load, to join entries with very few seconds with their parents, to keep the DB small. But there is no code in there that checks for date ranges or anything. This is in `nodedb.h:load()`.

One thing to try first, go to where your database files are stored (link from the start menu), and you should see dated backup files, depending on how often you restart your computer or PT. Look at the dates and the sizes. If there is a particular date at which the file became a lot smaller, then that's where your data got pruned?

If the DB never got smaller, it may mean that the data is still in the file, but simply always filtered for display. The code that relates to display and filtering is e.g. in `node.h:accumulate()`, in particular in `day_is_in_filter()` it checks if it is inside the date range. You could add code in there that remembers the earliest day that is the DB (smallest value for `o`), then compare that against `starttime` to understand more of what is going wrong.

**Comment by VelocityRa** [comment]
The DB did get smaller, there's a jump from 233KB -> 27KB, about a month ago. All entries after that are also small, slowly increasing (and some times decreasing, but not as much).
Should this pruning happen? I don't understand the point.

**Comment by aardappel** [comment]
With that big a jump in size another theory is that maybe at that date it wasn't able to load the old DB at all and started a new one from scratch.

If that is the case, the 2 DBs are non-overlapping, so you could use the merge DB feature (in the popup menu) to merge in the old one. Back up your DB files before attempting :P

Not sure what would cause that to happen though.

You could try loading the old DB to see if the data in there is not corrupted somehow.

**Comment by aardappel** [comment]
Yes, I suspect somehow the database read failed.. hard to tell why.

Tags are saved in the DB yes. It may have ignored them on merge..
